src/app/core/auth.service.spec.ts

```diff
@@ -1,6 +1,92 @@
+import { async, TestBed } from '@angular/core/testing';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { OAuthService } from 'angular-oauth2-oidc';
+import { MockOAuthService } from '../../testing/mocks';
+import { AuthService } from './auth.service';
 
 describe('AuthService', () => {
-  it('should pass', () => {
-    expect(true).toBeTruthy();
+  let service: AuthService;
+  let mockService: MockOAuthService;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule],
```

Comment:
There's a couple of errors on the console when running tests, due to the testing router running (asynchronously) into errors. I recommend doing:

```typescript
imports: [RouterTestingModule.withRoutes([
  { path: 'should-login', component: FakeComponent }
])],
```

with a `class FakeComponent { }`, or something similar. The errors in the log will then disappear.

Reply:
There were console warnings about router navigation outside the angular zone, so I wrapped in `zone.run` navigation inside event listener and it's fixed

Reply:
Hmm, I'm not sure if they were about the error being thrown outside of Angular? In fact, I think the error states that it _is_ inside Angular's scope, via a "ProxyZone":

> context.js:265 Unhandled Promise rejection: Cannot match any routes. URL Segment: 'should-login' ; Zone: ProxyZone ; Task: Promise.then ; Value: Error: Cannot match any routes. URL Segment: 'should-login'

I'm no expert on Angular's zone magic, but I don't think the changes with `zone.run(...)` are needed for this PR. The change using `.WithRoutes(...)` already fixes the error, and feels appropriate.

I'm happy if we open up a fresh issue, something like "_Run code XYZ within NgZone.run(...) because ..._" and be convinced of the need for that change. But I'd prefer to have a real (non-unit-test) scenario where this change (and added complexity) is needed. Hope that makes sense?

----

I'll tweak the PR with a small extra commit. If I can I will push it to the branch in the fork, otherwise I'll merge locally in a second and watch the master CI build to see if I did it correctly. Hope that's also ok!